packages/react/src/VideoPlayer/components/captions/Captions.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import clsx from 'clsx'
+import {Text} from '../../../Text'
+
+import styles from '../../VideoPlayer.module.css'
+
+type CaptionsProps = {
+  videoRef: React.RefObject<HTMLVideoElement>
+  disabled?: boolean
+  trackInformation?: TextTrackCueList
+}
+
+export const Captions = ({videoRef, disabled, trackInformation}: CaptionsProps) => {
+  const [caption, setCaption] = React.useState('')
+
+  React.useEffect(() => {
+    const compareAndSetCaption = () => {
+      if (trackInformation) {
+        for (let i = 0; i < trackInformation.length; i++) {
+          const cue = trackInformation[i]
+          if (
+            videoRef.current?.currentTime &&
+            videoRef.current.currentTime >= cue.startTime &&
+            videoRef.current.currentTime <= cue.endTime
+          ) {
+            setCaption(cue['text'])
+            break
+          } else {
+            setCaption('')
+          }
+        }
+      }
+    }
+
+    if (videoRef.current) {
+      videoRef.current.addEventListener('timeupdate', compareAndSetCaption)
+    }
+    return () => window.removeEventListener('timeupdate', compareAndSetCaption)
+  }, [videoRef, trackInformation])
+
+  return (
+    <div className={clsx(styles.VideoPlayer__captions, (disabled || !caption) && styles.VideoPlayer__captions__empty)}>
```

Comment:
Rather than using display: none to hide the element, could we short-circuit the function altogether outside of this return statement, to render `null` if `disabled || !caption`. Or do you need the DOM elements to always exist?

Reply:
Can we also use BEM conventions here please for consistency..

```suggestion
    <div className={clsx(styles.VideoPlayer__captions, (disabled || !caption) && styles.VideoPlayer__captions---empty)}>
```

Reply:
This has been updated in this SHA: https://github.com/primer/brand/pull/373/commits/3476819281a2ff27b50715bf57261befc6f1d0f5